### DISCUSSION TITLE
Dialogue box pops up asking to authenticate proxy on start always

### DISCUSSION THREAD

**Discussion by Stas1981ru** [qutebrowser/qutebrowser#5778] [Q&A] [answered]
I start QB with command line arguments to set proxy with proxy authentication.
I copyed QB from first machine to second machine.

First machine works fine.
Second machine always displays dialogue box pops up asking to authenticate on start (but proxy is set in command line arguments and I see it via :set content.proxy - it is correctly set with authentication)

Why proxy settings on second machine not applays on startup ?

**Comment by The-Compiler** [comment, marked as answer]
Not much I can say with so little info. What commandline arguments are you using? What versions of Qt, QtWebEngine and qutebrowser are running on those two machines?

**Reply by Stas1981ru** [reply]
Can I get this full info via command line in QB (like from crash) ?  Debian 10 buster, python3-pyqt5.* (5.11.3+dfsg-1+b3) from official repositories.

**Reply by Stas1981ru** [reply]
In photo you can see full info.
And as you see proxy is displays from :set content.proxy with authentication fields.
![Снимок экрана от 2020-10-08 19-30-48](https://user-images.githubusercontent.com/47504763/95487512-1b737900-099d-11eb-84a7-766623922467.png)
![Снимок экрана от 2020-10-08 19-29-47](https://user-images.githubusercontent.com/47504763/95487517-1c0c0f80-099d-11eb-8901-b1bebc5feaae.png)
![Снимок экрана от 2020-10-08 17-22-20](https://user-images.githubusercontent.com/47504763/95487521-1ca4a600-099d-11eb-8e16-89635249f00e.png)

**Reply by Stas1981ru** [reply]
in file  ./qutebrowser/browser/network/proxy.py in `queryProxy(self, query):` before line `return proxies` I has add
`print (proxies[0].password())`
At start it prints correct password.

QWebEnginePage.proxyAuthenticationRequired signal is emitted, but the QNetworkProxyFactory.setApplicationProxyFactory is set correctly ....

**Reply by The-Compiler** [reply]
Support for getting proxy authentication details from the URL automatically was added in Qt 5.13 - I'm guessing on your other machine where it works, you're using a newer Qt.

Also see #4278.

**Reply by Stas1981ru** [reply]
Yes, but on other machine Qt (5.15.0) is installed via pip3.
Debian 10 have not newer version in official repositories.
Why one machine works with pip install and other can't do it ? They all have Debian buster system.

**Reply by The-Compiler** [reply]
You probably installed qutebrowser differently then, your version info above clearly shows that you're using Qt 5.11 from Debian's packages. Like I already mentioned in #5777, if you want a newer Qt, check the relevant [install instructions](https://github.com/qutebrowser/qutebrowser/blob/master/doc/install.asciidoc#tox).
